### ISSUE TITLE
question about model.py

### ISSUE THREAD

**Issue by wooodswu** [lululxvi/deepxde#1259]
<img width="682" alt="image" src="https://user-images.githubusercontent.com/87766834/232689101-f204284c-0804-4763-accc-0cad363656f2.png">
Need your help! I have a question about this “losses_fn", except in the definition of “outputs_losses”, i cannot find this function. When I want to change the loss function, and using the following:

/*code

net = dde.maps.DeepONetCartesianProd( [101,30,30,101], [1,30,30,101], "relu", "Glorot normal") 

model2 =dde.Model(data, net) model2.compile("adam", lr=3e-4, loss=“mse", metrics=["l2 relative error"], )  

*/
 loss in dde.Model.compile()

by my understanding, is just the “loss_fn”(here is just mse) in the “losses_fn", and I cannot find the “losses_fn” in the whole package.

 My question: could I change this losses_fn? is this the loss function I need to define? Or in the somewhere of the model, where can I change the loss?

**Comment by kyouma** [comment]
You can pass a callable to the Model class: for example, _loss=torch.nn.functional.l1_loss_ instead of _loss='mse'_. And as far as I understand from the source code, a list of mixed identifiers/callables for the equations and the conditions also can be used.

**Comment by wooodswu** [comment]
> You can pass a callable to the Model class: for example, _loss=torch.nn.functional.l1_loss_ instead of _loss='mse'_. And as far as I understand from the source code, a list of mixed identifiers/callables for the equations and the conditions also can be used.

Could you tell me where the mixed identifiers/callables for the equation is and in which module? i have not found this in this package. Thanks a lot.

**Comment by kyouma** [comment]
Check [this `get()` function](https://github.com/lululxvi/deepxde/blob/e2f1259d0ea46ef7aaef4bfeaebaaa727c23e755/deepxde/losses.py#L56), which returns the loss function itself to the model during the initialization process. The check for callable is [on the line 70](https://github.com/lululxvi/deepxde/blob/e2f1259d0ea46ef7aaef4bfeaebaaa727c23e755/deepxde/losses.py#L70).
You can use this feature like this: `model.compile('adam', lr=1e-3, loss=torch.nn.functional.l1_loss)`.

**Comment by chuluya** [comment]
Hello, could you help solve the problems "Error: The network has multiple inputs, and saving such result han't been implemented." when I choose the DeepONet as my net. 
Thank you so much if you could help me!
![image](https://github.com/lululxvi/deepxde/assets/118424004/99dc3c4c-cc38-4389-8e23-acd49a20a7b5)

**Comment by kyouma** [comment]
> Hello, could you help solve the problems "Error: The network has multiple inputs, and saving such result han't been implemented." when I choose the DeepONet as my net. Thank you so much if you could help me! ![image](https://user-images.githubusercontent.com/118424004/238124320-99dc3c4c-cc38-4389-8e23-acd49a20a7b5.png)

As far as I understand, your network uses multiple input "streams", and DeepXDE has not implemented [saving](https://github.com/lululxvi/deepxde/blob/cda8547de38e3d429a067e384ee19d38d4311299/deepxde/utils/external.py#L326) and [plotting ](https://github.com/lululxvi/deepxde/blob/cda8547de38e3d429a067e384ee19d38d4311299/deepxde/utils/external.py#L241) input-output states for such models yet. In my opinion, you have several options:
1) not use the `deepxde.saveplot` function at all
- plot and save the `losshistory` using `deepxde.utils.external.plot_loss_history` and `deepxde.utils.external.save_loss_history` or your own code
- plot and save the input-output states (`train_state`) using your own code (in this case you need to examine the `train_state` structure, which is, as far as I remember, a combination of inputs and outputs; where the inputs are the internal area + external area points, and then external area points again for boundary conditions and also initial conditions area points, concatenated together; also check the `deepxde.utils.external.save_best_state` function, it is for single input "stream" networks)
2) modify your network to use a single input "steam" and continue to use the stock DeepXDE functions
